CMakeLists.txt

```diff
@@ -306,6 +306,16 @@ set(BOUT_SOURCES
   ${CMAKE_CURRENT_BINARY_DIR}/include/bout/version.hxx
   )
 
+
+ADD_CUSTOM_COMMAND( OUTPUT ${CMAKE_CURRENT_SOURCE_DIR}/src/field/generated_fieldops.cxx
+  COMMAND python gen_fieldops.py --filename generated_fieldops.cxx.tmp
+  COMMAND clang-format generated_fieldops.cxx.tmp -i
```

Comment:
We can also add `FindClangFormat` from [here](https://github.com/ttroy50/cmake-examples/blob/64bd54a1a11dcc83c44280d3c255218ffc1b2d66/04-static-analysis/clang-format/cmake/modules/FindClangFormat.cmake) and do a similar thing for `clang-format`.